src/odemis/acq/align/tdct.py

```diff
@@ -0,0 +1,146 @@
+        z = prev_z
+        x, y = prev_x, prev_y
+
+    return z
+
+
+def run_tdct_correlation(fib_coords: numpy.ndarray,
+                           fm_coords: numpy.ndarray,
+                           poi_coords: numpy.ndarray,
+                           fib_image: model.DataArray,
+                           fm_image: model.DataArray,
+                           path: str) -> Dict[str, Tuple[float, float]]:
+    """Run 3DCT Multi-point correlation between FIB and FM images.
+    :param fib_coords: the FIB coordinates (x, y, z)
+    :param fm_coords: the FM coordinates (x, y, z)
+    :param poi_coords: the point of interest coordinates
+    :param fib_image: the FIB image
+    :param fm_image: the FM image
+    :param path: the path to save the results
+    :return: the correlation results (including poi, transformations, and errors)
+    """
+
+    # get rotation center
+    halfmax_dim = int(max(fm_image.shape) * 0.5)
+    rotation_center = (halfmax_dim, halfmax_dim, halfmax_dim)
+
+    # get fib pixel size (meters)
+    fib_pixel_size = fib_image.metadata[model.MD_PIXEL_SIZE][0]
+
+    # fib image shape minus metadata, fib_pixelsize (microns), fm_image_shape
+    image_props = [fib_image.shape, fib_pixel_size * 1e6, fm_image.shape]
+
+    assert fib_image.ndim == 2, "FIB Image must be 2D"
+    assert fm_image.ndim == 3, "FM Image must be 3D"
+    assert fib_pixel_size is not None, "FIB Pixel Size must be set"
+    assert rotation_center is not None, "Rotation Center must be set"
+    assert isinstance(rotation_center, tuple), "Rotation Center must be a tuple"
+    assert len(rotation_center) == 3, "Rotation Center must have 3 values"
+
+    logging.debug(
+        f"Running 3DCT correlation with FIB image shape: {fib_image.shape}, FM image shape: {fm_image.shape}"
+    )
+
+    # run correlation
+    correlation_results = run_correlation(
+        fib_coords=fib_coords,
+        fm_coords=fm_coords,
+        poi_coords=poi_coords,
+        image_props=image_props,
+        rotation_center=rotation_center,
+        path=path,
+        fib_image_filename="FIB-Image",
+        fm_image_filename="FM-Image",
+    )
+
+    return correlation_results
+
+def get_poi_coordinate(correlation_results: dict) -> Tuple[float, float]:
+    """Get the the point of interest coordinate from correlation data
+    and convert from micrometers to meters in the microscope image coordinate system.
+    :param correlation_results: the correlation results
+    :return: the point of interest coordinate in meters
+    """
+    # get the point of interest coordinate (in microscope coordinates, in metres)
+    poi_coord = correlation_results["output"]["poi"][0]["px_um"]
+    poi_coord = (poi_coord[0] * 1e-6, poi_coord[1] * 1e-6)
+    return poi_coord
+
+def parse_3dct_yaml_file(path: str) -> Tuple[float, float]:
+    """Parse the 3DCT yaml file and extract the point of interest (POI)
```

Comment:
Is that function going to be useful? Why would we have a YAML file generated by 3DCT?

Reply:
This is so you can load the correlated point from external use of 3dct for other methods that odemis doesn't support. E.g. right now multi-point isn't available in odemis but is in 3dct. So users could technically use whatever correlation method they want, and then import the position into odemis and use the rest of the workflow